packages/grpc-js-core/src/channel.ts

```diff
@@ -204,11 +207,12 @@ export class Http2Channel extends EventEmitter implements Channel {
   private startHttp2Stream(
       methodName: string, stream: Http2CallStream, metadata: Metadata) {
     let finalMetadata: Promise<Metadata> =
-        stream.filterStack.sendMetadata(Promise.resolve(metadata));
+        stream.filterStack.sendMetadata(Promise.resolve(metadata.clone()));
     Promise.all([finalMetadata, this.connect()])
       .then(([metadataValue]) => {
         let headers = metadataValue.toHttp2Headers();
         headers[HTTP2_HEADER_AUTHORITY] = this.authority.hostname;
+        headers[HTTP2_HEADER_USER_AGENT] = `grpc-node/${clientVersion}`;
```

Comment:
I think this should be `grpc-node-js` or something like that, so that it's clearly a variant. Also, I think it would be good to pipe through the user agent channel arguments. There are actually two of them: `grpc.primary_user_agent` gets put at the beginning, and `grpc.secondary_user_agent` goes at the end.

Reply:
I added them and also moved some code so that `options` doesn't get modified. I didn't see code to add `secondary_user_agent` in grpc-native-core though (I might be mistaken).